### DISCUSSION TITLE
Adding NLP support to the discord bot

### DISCUSSION THREAD

**Discussion by janaSunrise** [EddieHubCommunity/support#656] [Ideas]
This is suggested, as it gives suggestions without understanding the context or, what is happening around. This can be solved by processing language, and knowing what it actually means.

Like, it(the bot) gives he/she for referring to the bots, or something else that doesn't match up with the actual context because of the hard coding. using NLP it can understand what's actually going on and provide suggestions that actually match up to the context

Here are some good resources I found for NLP:

- [https://www.kommunicate.io/blog/nlp-libraries-node-javascript/](https://www.kommunicate.io/blog/nlp-libraries-node-javascript/)
- [https://medium.com/@daffl/natural-language-processing-and-machine-learning-in-javascript-249181a3b721](https://medium.com/@daffl/natural-language-processing-and-machine-learning-in-javascript-249181a3b721)
- [https://openbase.io/categories/js/best-nodejs-nlp-libraries](https://openbase.io/categories/js/best-nodejs-nlp-libraries)

Here are some good libraries for NLP:

- [Tensorflow](https://tensorflow.org/)
- [Brain JS](https://brain.js.org)

And also maybe support so that it tracks messages to catch up the context. and also helps in logging any issues too. This will be a good improvement and achievement in my opinion, and will give suggestion only when it is required.

**Comment by eddiejaoude** [comment]
Thanks for sharing this 👍

**Comment by janaSunrise** [comment]
I would love to see this change. It needs a fix, and it'd be more appealing and understandable.

**Reply by Vyvy-vi** [reply]
Would adding NLP to the bot become resource-intensive, or would it be alright?

**Reply by janaSunrise** [reply]
It would be the same until you implement a lot of deep learning algorithms. Or if you want to do that, best way would be to export the model and then utilize it, and it's gonna work normally.
